add V4.0 tidb binlog restriction (#5128)
pingcap/docs: tidb-binlog/tidb-binlog-overview.md

## Notes

* TiDB Binlog is incompatible with the following feature introduced in TiDB v4.0.7 and they cannot be used together:

>     - TiDB system variable [tidb_enable_amend_pessimistic_txn](/system-variables.md#tidb_enable_amend_pessimistic_txn-new-in-v407): The two features have compatibility issues. Using them together will cause correctness issues with inconsistent replication data of TiDB Binlog.

Reviewer: ```suggestion
    - TiDB system variable [tidb_enable_amend_pessimistic_txn](/system-variables.md#tidb_enable_amend_pessimistic_txn-new-in-v407): The two features have compatibility issues. Using them together might cause the issue that TiDB Binlog replicates inconsistent data.
```
